src/core/Akka/Util/AppVersion.cs

```diff
@@ -32,20 +34,50 @@ namespace Akka.Util
     public class AppVersion : IComparable<AppVersion>, IEquatable<AppVersion>
     {
         public static readonly AppVersion Zero = new AppVersion("0.0.0");
+
+        // made internal for testing purposes
+        internal const string AssemblyVersionMarker = "assembly-version";
         private const int Undefined = 0;
 
-        private int[] numbers = Array.Empty<int>();
-        private string rest = "";
+        private int[] _numbers = Array.Empty<int>();
+        private string _rest = "";
 
         [JsonConstructor]
         internal AppVersion(string version)
         {
-            this.Version = version;
+            Version = version;
         }
 
         public static AppVersion Create(string version)
         {
-            var v = new AppVersion(version);
+            // check to see if we're going to use the assembly-version
+            if (version.Equals(AssemblyVersionMarker, StringComparison.InvariantCultureIgnoreCase))
+            {
+                return AppVersionFromAssemblyVersion();
+            }
+
+            var v2 = new AppVersion(version);
+            return v2.Parse();
+        }
+
+        /// <summary>
+        /// INTERNAL API
+        /// </summary>
+        /// <remarks>
+        /// Internal for testing purposes only.
+        /// </remarks>
+        internal static AppVersion AppVersionFromAssemblyVersion()
+        {
+            // user hasn't specified AppVersion in HOCON
+            // try looking it up via assembly
+            var entryAssembly = Assembly.GetEntryAssembly();
+
+            // if the entryAssembly is null (which can happen when we're called from unmanaged code)
+            // then fall back to the executing assembly for the version number.
+            var targetAssembly = entryAssembly ?? Assembly.GetExecutingAssembly();
```

Comment:
Grab the `EntryAssembly` preferably. If this value is `null`, which can happen when called from unmanaged code (this happens when the XUnit test runner is executing from within visual studio, for instance) then fallback to the `ExecutingAssembly` version. Shouldn't happen in most Akka.NET applications, which tend to be launched as 100% managed processes.